scripts/create-latest-json.sh

```
@@ -0,0 +1,264 @@
+
+  # Prepare the directory and extract files
+  prepare_directory_files "$artifact_path" "$platform_key"
+  # Check if the directory was created successfully
+  if [[ ! -d "$BASE_DIR/$platform_key" ]]; then
+    echo "Error: Failed to create directory $BASE_DIR/$platform_key"
+    return 1
+  fi
+
+  # Find the main file and its signature
+  echo "Finding main file and signature for $platform_key..."
+  local main_file=$(find "$BASE_DIR/$platform_key" -type f -name "*.AppImage" -o -name "*.msi" -o -name "*.tar.gz" | head -n 1)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)